app/authentication/session_management.py

```diff
@@ -84,12 +39,21 @@ def store_user_id(self, user_id):
             logger.debug("Found eq_session_id %s in session", eq_session_id)
             eq_session = self._get_object(eq_session_id)
             logger.debug("Loaded object eq session %s", eq_session)
-        logger.debug("About to commit to database")
-        db_session.add(eq_session)
-        db_session.commit()
-        logger.debug("Committed")
+            logger.debug("About to commit to database")
+
+        try:
```

Comment:
Definitely a better way of doing it.

The other side of this is that we could start a transaction at the beginning of the flask request and commit/rollback at the end but I guess it depends what we class as a unit of work. At the moment there is only one place where we do more than one DML action on login.